logic/auth.js

```diff
@@ -91,6 +91,7 @@ async function changePassword(currentPassword, newPassword, jwt) {
             // wait for lnd to boot up
             if (error.response.status === constants.STATUS_CODES.BAD_GATEWAY) {
                 await sleepSeconds(1);
+                attempt--;
```

Comment:
I wouldn't do that because, as you already said, you risk getting in an infinite loop.

You could just
a) make the max attempts higher because the lnd bootup is the only check that even uses it, everything else is (or should, see other comment) an error resulting in immediate loop exit.
b) make the sleep longer, 5 seconds is IMO a sensible interval for this (worst case it takes 4 seconds longer than needed)

Combining these would probably be the best solution if this were to be implemented.

Reply:
Yeah, agree on not using `attempt--`.

> you can just say the `const MAX_ATTEMPTS` equals the max timeout in seconds

The max timeout is probably going to be way higher than the max attempts, since an attempt can last longer than 1s as it makes synchronous API and disk read/write calls (and there's no way to know that before hand).